lib/disposable.dart

```diff
@@ -143,10 +145,13 @@ typedef Future<dynamic> Disposer();
 /// without explicit reference to [Disposable]. To do this, we use
 /// composition to include the [Disposable] machinery without changing
 /// the public interface of our class or polluting its lifecycle.
+///
+/// Implementing the [DisposableManager] interface is not required.
 class Disposable implements _Disposable, DisposableManager {
   Completer<Null> _didDispose = new Completer<Null>();
   bool _isDisposing = false;
   List<_Disposable> _internalDisposables = [];
+  Set<Future> _blockingFutures = new Set<Future>();
```

Comment:
I don't think there's anything wrong with using `Future<dynamic>` in this case, it is still correctly typed since we ignore the value it resolves to. I can be more explicit about it though.